src/types/composer.ts

```diff
@@ -0,0 +1,673 @@
+import algosdk from 'algosdk'
+import { sendAtomicTransactionComposer } from '../transaction/transaction'
+import { TransactionSignerAccount } from './account'
+import { AlgoAmount } from './amount'
+
+export type CommonTxnParams = {
+  /** The address sending the transaction */
+  sender: string
+  /** The function used to sign transactions */
+  signer?: algosdk.TransactionSigner | TransactionSignerAccount
+  /** Change the signing key of the sender to the given address */
+  rekeyTo?: string
+  /** Note to attach to the transaction*/
+  note?: Uint8Array | string
+  /** Prevent multiple transactions with the same lease being included within the validity window */
+  lease?: Uint8Array | string
+  /** The transaction fee. In most cases you want to use `extraFee` unless setting the fee to 0 to be covered by another transaction */
+  staticFee?: AlgoAmount
+  /** The fee to pay IN ADDITION to the suggested fee. Useful for covering inner transaction fees */
+  extraFee?: AlgoAmount
+  /** Throw an error if the fee for the transaction is more than this amount */
+  maxFee?: AlgoAmount
+  /** How many rounds the transaction should be valid for */
+  validityWindow?: number
+  /**
+   * Set the first round this transaction is valid.
+   * If left undefined, the value from algod will be used.
+   * Only set this when you intentionally want this to be some time in the future
+   */
+  firstValidRound?: bigint
+  /** The last round this transaction is valid. It is recommended to use validityWindow instead */
+  lastValidRound?: bigint
+}
+
+export type PayTxnParams = CommonTxnParams & {
+  /** That account that will receive the ALGO */
+  receiver: string
+  /** Amount to send */
+  amount: AlgoAmount
+  /** If given, close the sender account and send the remaining balance to this address */
+  closeRemainderTo?: string
+}
+
+export type AssetCreateParams = CommonTxnParams & {
+  /** The total amount of the smallest divisible unit to create */
+  total: bigint
+  /** The amount of decimal places the asset should have */
+  decimals?: number
+  /** Whether the asset is frozen by default in the creator address */
+  defaultFrozen?: boolean
+  /** The address that can change the manager, reserve, clawback, and freeze addresses */
+  manager?: string
+  /** The address that holds the uncirculated supply */
+  reserve?: string
+  /** The address that can freeze the asset in any account */
+  freeze?: string
+  /** The address that can clawback the asset from any account */
```

Comment:
Should we include something in these comments to explain that setting them to "" will disable the associated behaviour for the asset?

Reply:
Done in cd4f67278696c74f9aeb2a9852c56f8f774b642a